### PR TITLE
adds a json search-inside endpoint

### LINKED ISSUE #3532: is there a JSON API for inside search results?
### Question

Is there a JSON API URL for inside-search requests? I could not find it in the API documentation. Adding ".json" to the end of "inside" does not work here.

The HTML search URL is: [https://openlibrary.org/search/inside?q=%22stocked+with+wine-related+goods%22&mode=ebooks&has_fulltext=true](https://openlibrary.org/search/inside?q=%22stocked+with+wine-related+goods%22&mode=ebooks&has_fulltext=true)

I would have exepected this to work for JSON output: [https://openlibrary.org/search/inside.json?q=%22stocked+with+wine-related+goods%22&mode=ebooks&has_fulltext=true](https://openlibrary.org/search/inside.json?q=%22stocked+with+wine-related+goods%22&mode=ebooks&has_fulltext=true)

Thanks!

### PR DESCRIPTION
**Pull request by mekarpeles**
Closes #3532

### CODE REVIEW

File: openlibrary/plugins/inside/code.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 from time import time
 
-import simplejson
+import json
```
Comment (cclauss) [inline openlibrary/plugins/inside/code.py:3]: Yeah!!  Let’s keep doing this!

**Comment by cclauss** [comment]
`make lint-diff` issues https://travis-ci.com/github/internetarchive/openlibrary/jobs/430449819#L581-L586

File: openlibrary/core/fulltext.py [hunk 2]
```diff
@@ -49,7 +49,7 @@ def fulltext_search(q, page=1, limit=100):
         for ed in editions:
             if ed.ocaid in ocaids:
                 idx = ocaids.index(ed.ocaid)
-                ia_results['hits']['hits'][idx]['edition'] = ed
+                ia_results['hits']['hits'][idx]['edition'] = format_book_data(ed) if js else ed
```
Comment (mekarpeles) [inline openlibrary/core/fulltext.py:52]:
```suggestion
                ia_results['hits']['hits'][idx]['edition'] = \
                    format_book_data(ed) if js else ed
```

File: openlibrary/plugins/inside/code.py [hunk 3]
```diff
@@ -27,3 +27,17 @@ def GET(self):
                                page=page, results_per_page=RESULTS_PER_PAGE)
         page.v2 = True  # page is mobile-first
         return page
+
+
+class search_inside_json(delegate.page):
+    path ="/search/inside"
```
Comment (mekarpeles) [inline openlibrary/plugins/inside/code.py:33]:
```suggestion
    path = "/search/inside"
```

File: openlibrary/core/fulltext.py [hunk 4]
```diff
@@ -49,8 +49,8 @@ def fulltext_search(q, page=1, limit=100):
         for ed in editions:
             if ed.ocaid in ocaids:
                 idx = ocaids.index(ed.ocaid)
-                ia_results['hits']['hits'][idx]['edition'] = ed
+                ia_results['hits']['hits'][idx]['edition'] = \
+                    format_book_data(ed) if js else ed
```
Comment (cclauss) [inline openlibrary/core/fulltext.py:53]:
```suggestion
                ia_results['hits']['hits'][idx]['edition'] = (
                    format_book_data(ed) if js else ed
                )
```
Let's avoid backslashes in Python code as discussed in PEP8.

**Review by cclauss — APPROVED** [review]
RSLGTM

**Pull request merged**